### PR TITLE
Backport 8307850 to jdk11

### PR DESCRIPTION
**Pull request by pshipton**
8307850: update for deprecated sprintf for jdk.jdi
https://github.com/ibmruntimes/openj9-openjdk-jdk17/commit/d1acd4d

### CODE REVIEW

File: src/jdk.jdi/share/native/libdt_shmem/SharedMemoryTransport.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1999, 2004, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1999, 2024, Oracle and/or its affiliates. All rights reserved.
```
Comment (keithc-ca) [inline src/jdk.jdi/share/native/libdt_shmem/SharedMemoryTransport.c:2]: The second date is 2023 in jdk17; I suggest it should be the same here.

File: src/jdk.jdi/share/native/libdt_shmem/shmemBase.c [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1999, 2018, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1999, 2024, Oracle and/or its affiliates. All rights reserved.
```
Comment (keithc-ca) [inline src/jdk.jdi/share/native/libdt_shmem/shmemBase.c:2]: The second year is 2020 in jdk17: I suggest it should be the same here.

**Review by keithc-ca — CHANGES_REQUESTED** [review]
Please update the commit message and the description to use the public URL for the referenced commit: https://github.com/ibmruntimes/openj9-openjdk-jdk17/commit/d1acd4d8943b910935523359ee6d9dd22a3a471b.

File: src/jdk.jdi/share/native/libdt_shmem/shmemBase.c (hunk 2 above)
Comment (pshipton) [reply src/jdk.jdi/share/native/libdt_shmem/shmemBase.c]: Updated them both.

**Comment by keithc-ca** [comment]
> Please update the commit message and the description to use the public URL for the referenced commit: [ibmruntimes/openj9-openjdk-jdk17@d1acd4d](https://github.com/ibmruntimes/openj9-openjdk-jdk17/commit/d1acd4d8943b910935523359ee6d9dd22a3a471b).

**Comment by pshipton** [comment]
Fixed it.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**